src/layout/core/borders/container_draw.rs

```diff
@@ -33,21 +33,96 @@ impl ContainerDraw {
+                let tab_width = w / children.titles.len() as f64;
+
+                for (i, title) in children.titles.iter().enumerate() {
+                    let active = children.index == i;
+
+                    let title_color = if active { title_color }
+                        else { Borders::default_title_color() };
+
+                    let title_font_color = if active { title_font_color }
+                        else { Borders::default_title_font_color() };
+
+                    let x = x + tab_width*i as f64;
+                    let title_x = title_x + tab_width*i as f64;
+
+                    // Draw background of title bar
+                    self.base.set_source_rgb(1.0, 0.0, 0.0);
+                    self.base.set_color_source(title_color);
+                    self.base.rectangle(x, 0.0, tab_width, title_size);
+                    self.base = try!(self.base.check_cairo());
+                    self.base.fill();
+                    self.base = try!(self.base.check_cairo());
+
+                    // Draw title text
+                    self.base.move_to(title_x, title_y);
+                    self.base = try!(self.base.check_cairo());
+                    self.base.set_color_source(title_font_color);
+                    self.base = try!(self.base.check_cairo());
+                    self.base.show_text(title.as_str());
+                    self.base = try!(self.base.check_cairo());
+                }
+            }
+            (Some(Layout::Stacked), Some(children)) => {
+                for (i, title) in children.titles.iter().enumerate() {
+                    let active = children.index == i;
+
+                    let title_color = if active { title_color }
+                        else { Borders::default_title_color() };
+
+                    let title_font_color = if active { title_font_color }
+                        else { Borders::default_title_font_color() };
+
+                    let y = title_size*i as f64;
+                    let title_y = title_y as f64 + y;
+
+                    // Draw background of title bar
+                    self.base.set_source_rgb(1.0, 0.0, 0.0);
+                    self.base.set_color_source(title_color);
+                    self.base.rectangle(x, y, w, title_size);
+                    self.base = try!(self.base.check_cairo());
+                    self.base.fill();
+                    self.base = try!(self.base.check_cairo());
+
+                    // Draw title text
+                    self.base.move_to(title_x, title_y);
+                    self.base = try!(self.base.check_cairo());
+                    self.base.set_color_source(title_font_color);
+                    self.base = try!(self.base.check_cairo());
+                    self.base.show_text(title.as_str());
+                    self.base = try!(self.base.check_cairo());
+                }
+            }
+            _ => {
+                let title: String = self.inner().title().into();
```

Comment:
This is an unnecessary copy. You get borrowck errors if you just return the `&str` here, but you can avoid them by just changing line 121 to be:

```rust
self.base.show_text(self.inner().title());
```

and not bother binding it before that at all.